hotxlfp/formulas/logic.py

```diff
@@ -16,7 +17,7 @@ def AND(*args):
 
 @dispatcher.register_for('IF')
 def IF(test, then, otherwise):
-    return then if test else otherwise
+    return np.where(test, then, otherwise)
```

Comment:
what happens if a non-numpy number type is passed in here? do we return a numpy array for the entire calculation now?

Reply:
Yes, this will return a 0 dimensional numpy array. I think this should be okay, but I can test this when I do integration.

Reply:
Okay, this seems fine for now but we should discuss how to handle this in the more general case live. We'll need to figure out how to make this run smoothly with numpy, torch, and scalar values